### ISSUE TITLE
As an IO I want to upload documents to forms

### ISSUE THREAD

**Issue by BCerki** [bcgov/cas-registration#325]
#### Description:

We decided to split document upload out of the forms tickets. This ticket is about adding it. As of January 2024, only one document will be uploaded, the Statutory Declaration.

#### Acceptance Criteria:
Wireframe: https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=4903%3A986&mode=design&t=rhr96fvLC4QihbUA-1

Given I am logged in to the Registration app with my Business BCeID account
When I navigate to the Operation form
Then I can add the statutory declaration document to the form

Given I am logged in to the Registration app with my IDIR account
When I navigate to the page for a specific Operation
Then I can view the file that has been uploaded for statutory declaration

~~ Given I am logged in to the Registration app with my Business BCeID account and I have been granted "admin" role for my operator
When I navigate to the Operation (facility) form
Then I can add documents to the form (both when creating and editing)

Given I am logged in to the Registration app with my Business BCeID account and I have been granted "admin" role for my operator
When I navigate to the  OperatOR form
Then I can add documents to the form (both when creating and editing)

Given I am logged in to the Registration app with my IDIR account and I have been granted the role of a CAS internal staff member
When I navigate to the page for a specific Operation/Operator
Then I can view the files that have been uploaded for that Operation/Operator ~~

#### Development Checklist:

- [ ] Add document handling to the existing operation forms (django ninja has docs about document handling but it  seems to be if you're using views instead of React)
- [ ] Documents fields are already in the models, but may need to made mandatory (remove blank=True) if they are required (see spreadsheet Required column)
- [ ] Check if the files expires
- [ ] Determine when is best to upload (as soon as the user uploads, or on submit of the form)
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

- Blocked by #332 
- Blocked by #604

**Comment by BCerki** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Moose-Sprint-Board-60d216175a2689000ec774d0/issues/bcgov/cas-registration/325?planning-poker) with Zenhub @dleard @joshgamache @JoshLarouche @mikevespi @Sepehr-Sobhani @shon-button

**Comment by Dianadec** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Moose-Sprint-Board-60d216175a2689000ec774d0/issues/bcgov/cas-registration/325?planning-poker) with Zenhub @marcellmueller

**Comment by Dianadec** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Moose-Sprint-Board-60d216175a2689000ec774d0/issues/bcgov/cas-registration/325?planning-poker) with Zenhub @andrea-williams

**Comment by BCerki** [comment]
For when we pick this back up again:
- PR that implements boundary map upload: https://github.com/bcgov/cas-registration/pull/532
- PR of document exploration: https://github.com/bcgov/cas-registration/pull/507

**Comment by Dianadec** [comment]
This ticket AC may need refining according to the new wireframe: https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=4903%3A986&mode=design&t=rhr96fvLC4QihbUA-1

**Comment by Dianadec** [comment]
Please add your estimates here: https://planningpokeronline.com/zz1T5DgYXidDPQtTKck2/ @dleard  @BCerki @shon-button @marcellmueller @Sepehr-Sobhani @andrea-williams

**Comment by Dianadec** [comment]
Confirming whether or not we want to hard-delete updated attachments. cc @dleard  @hannavovk

**Comment by hannavovk** [comment]
Looking great and seems to working properly. Not sure that #799 is needed. @BCerki @marcellmueller
